Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectBuilder/MSBuildProjectBuilder.cs

```diff
@@ -155,20 +157,28 @@ public static bool BuildProjects(this IReadOnlyCollection<MSBuildProjectReferenc
                     {
                         int completedProjects = 0;
                         string progressMessage = string.Empty;
+                        Match adoAuthenticationMatch = null;
                         var cancellationTokenSource = new CancellationTokenSource();
 
                         DisplayProgress();
 
                         Task<bool> buildTask = MSBuildProjectBuilder.BuildProjectsAsync(
                             msBuildProjectReferences,
-                            $"{arguments} -v:minimal",
+                            $"{arguments} -v:minimal -p:NuGetInteractive=true",
                             new DelegateProgress<(int completedProjects, (string progressMessage, ProgressMessageType progressMessageType) progressUpdate)>(report =>
                             {
                                 if (report.progressUpdate.progressMessageType != ProgressMessageType.Information)
                                 {
                                     MSBuildProjectBuilder.LogProgressMessage(report.progressUpdate.progressMessage, report.progressUpdate.progressMessageType);
                                 }
+
                                 (completedProjects, progressMessage) = (report.completedProjects, report.progressUpdate.progressMessage);
```

Comment:
>(completedProjects, progressMessage) [](start = 32, length = 36)

Not introduced by this change (so feel free to consider out of scope), but it looks like this delegate is making assignments to various variables on an arbitrary thread, but then the display loop below is reading/manipulating them all on the main thread?  Maybe some simplistic locking?

Alternatively if this is actually on the main thread could the display logic be rolled up to here?

Reply:
The delegate does run on a background thread, and it does assign to these variables, but the assignment (to each variable) is atomic, and the variables being displayed are done so at 30fps, so even if they are out of sync for a frame, it wouldn't be noticeable. Given this, I would lean towards just leaving it as is.